frontend/services/index.ts

```diff
@@ -1,7 +1,72 @@
-import axios, { isAxiosError, ResponseType as AxiosResponseType } from "axios";
+import axios, {
+  isAxiosError,
+  ResponseType as AxiosResponseType,
+  AxiosProgressEvent,
+} from "axios";
 import URL_PREFIX from "router/url_prefix";
 import { authToken } from "utilities/local";
 
+export const sendRequestWithProgress = async ({
```

Comment:
it would be nice if we could just use `sendRequest` method instead of create a new one. Would the refactoring of the arguments to be an object be a lot of work? I'm thinking we leave the first 3, and the 4th being an options argument to make the refactoring smaller.

Reply:
I thought about that but to keep the scope of the PR manageable, I opted for this instead. We only need this functionality in a couple places currently so if we do decide to update the basic `sendRequest` function with more params, it should be easy enough to fold this in then.